uqcsbot/scripts/crates.py

```diff
@@ -0,0 +1,561 @@
+    if args.user.isdigit():
+        params['user_id'] = args.user
+    elif args.user:
+        user_id = get_user_id(args.user)
+        if user_id == -1:
+            bot.post_message(channel, f'The username {args.user} could not be resolved')
+            return
+
+        params['user_id'] = str(user_id)
+
+    search_result = get_crates_search_results(channel, args.search, params)
+    if search_result is None:
+        return
+
+    crates, total = search_result
+
+    # No crates at all were found
+    if not crates:
+        bot.post_message(channel, "No crates were found")
+        return
+
+    # The beginning of the formatted response
+    blocks = [
+        {
+            'type': 'section',
+            'text': {
+                'type': 'mrkdwn',
+                'text': f'*Showing {min(args.limit, total)} of {total} results*'
+            },
+        },
+        {
+            'type': 'divider'
+        }
+    ]
+
+    # Iterate over all of the crates or until limit is reached. Whichever comes first.
+    page = 1
+    remaining = args.limit
+    while remaining > 0:
+        amt = range(0, min(len(crates), remaining))
+        for index in amt:
+            crate = crates[index]
+            blocks.extend(get_crate_blocks(crate))
+            remaining -= 1
+
+        page += 1
+        search_result = get_crates_search_results(channel, args.search, params, page)
+        if search_result is None or not search_result[0]:
+            break
+
+        crates, _ = search_result
+
+    bot.post_message(channel, '', blocks=blocks)
+
+
+def get_category_page(channel: Channel, sort: str, page: int) -> Tuple[Optional[List[str]], int]:
+    """
+    Returns all the names of all categories from a page of the response
+    :param channel: The channel to post any errors to
+    :param sort: The order to sort by. One of "crates" or "alpha"
+    :param page: The page number to get the categories from
+    :return: A tuple containing a list of category names (or None on error) and the total number of categories
+    """
+    # Get the categories
+    url = BASE_URL + '/categories'
+    response = requests.get(url, {'sort': sort, 'page': page})  # type: ignore
+
+    if response.status_code != requests.codes.ok:
+        bot.post_message(channel, 'There was a problem getting the list of categories')
+        return None, 0
+
+    # Convert the json response
+    response_data = json.loads(response.content)
+
+    raw_categories = response_data.get('categories')
+    total = response_data.get('meta', {}).get('total', 0)
+
+    # Get the category names
+    categories = [cat.get('name') if 'name' in cat else cat.get('id', '') for cat in raw_categories]
+
+    return categories, total
+
+
+def display_all_categories(channel: Channel, args: CategorySearch):
+    """Displays just the names of all the categories in one big list"""
+    cats, tot = get_category_page(channel, args.sort, 1)
```

Comment:
Please avoid using shorthand for variables where possible, it reduces clarity and we'd prefer comprehensibility over concision.